common/reflect_resource_plugin_framework.go

```diff
@@ -87,21 +93,52 @@ func tfSdkToGoSdkSingleField(srcField reflect.Value, destField reflect.Value, sr
 				addToForceSendFields(srcFieldName, forceSendFieldsField)
 			}
 		case types.String:
-			destField.SetString(v.ValueString())
-			if !v.IsNull() {
-				addToForceSendFields(srcFieldName, forceSendFieldsField)
+			if destField.Type().Name() != "string" {
+				// This is the case for enum.
+
+				// Find the Set method
+				destVal := reflect.New(destField.Type())
+				setMethod := destVal.MethodByName("Set")
+				if !setMethod.IsValid() {
+					panic(fmt.Sprintf("set method not found on enum type: %s", destField.Type().Name()))
+				}
+
+				// Prepare the argument for the Set method
+				arg := reflect.ValueOf(v.ValueString())
+
+				// Call the Set method
+				result := setMethod.Call([]reflect.Value{arg})
+				if len(result) != 0 {
+					if err, ok := result[0].Interface().(error); ok && err != nil {
+						panic(err)
+					}
+				}
+				destField.Set(destVal.Elem())
+			} else {
+				destField.SetString(v.ValueString())
+				if !v.IsNull() {
+					addToForceSendFields(srcFieldName, forceSendFieldsField)
+				}
 			}
 		case types.List:
 			panic("types.List should never be used, use go native slices instead")
 		case types.Map:
 			panic("types.Map should never be used, use go native maps instead")
 		default:
+			if srcField.IsZero() {
+				// Skip nils
```

Comment:
Skip Zeros probably would be better as comment as we are checking for zero values.

Reply:
Yeah, good call, let me update it